packages/store-cache/src/utils.ts

```diff
@@ -26,43 +34,42 @@ import { getAbiTypeDefaultValue } from "@latticexyz/schema-type/deprecated";
  * }
  * @returns Transaction
  */
-export function set<C extends StoreConfig, T extends keyof C["tables"] = keyof C["tables"]>(
+export async function set<C extends StoreConfig, T extends keyof C["tables"] = keyof C["tables"]>(
   config: C,
-  client: TupleDatabaseClient,
+  client: AsyncTupleDatabaseClient,
   namespace: C["namespace"],
   table: T & string,
   key: Key<C, T>,
   value: Partial<Value<C, T>>,
   options?: SetOptions
-) {
+): Promise<void> {
```

Comment:
I couldn't get this to work well with the conditional transaction. I tried something like:
```ts
SetOptions['transaction'] extends undefined ? Promise<void> : Promise<AsyncTupleRootTransactionApi>
```
but no luck. 

Also seems fine to not return the transaction, since you'll already have a reference to it if passed into `options`.

Reply:
yep i think that makes sense - if you want to stack transactions you have to pass it in, otherwise the transaction created in the function call is immediately committed and not useful outside the function